docs/src/man/joins_new.md

```diff
@@ -0,0 +1,879 @@
+- `rightjoin`: The output contains rows for values of the key that exist in the second (right) argument 
+  to `join`, whether or not that value exists in the first (left) argument.
+- `outerjoin`: The output contains rows for values of the key that exist in the first (left) or second 
+  (right) argument to `join`.
+
+## Anti Join
+
+The output contains rows for values of the key that exist in the first (left) but not the second 
+(right) argument to join. As with semi joins, output is restricted to columns from the first (left) 
+argument.
+
+Finally, we can keep only rows that *don't* match with `antijoin`:
+
+```jldoctest dataframe
+julia> antijoin(german, risk1, on = :id)
+4×5 DataFrame
+ Row │ id     Age    Sex     Job    Housing
+     │ Int64  Int64  String  Int64  String
+─────┼──────────────────────────────────────
+   1 │     0     67  male        2  own
+   2 │     1     22  female      2  own
+   3 │     2     49  male        1  own
+   4 │     3     45  male        2  free
+```
+
+## Semi Join
+
+It is like an inner join, but output is restricted to columns from the first (left) argument to join.
+
+Now, let’s say we want to look at german’s data for whom we have risk data, but we don’t actually 
+want to have all the columns from the second table. That’s what `semijoin` does. It gives us the same 
+rows as an `innerjoin`, but doesn’t add any columns from the 2nd table:
+
+```jldoctest dataframe
+julia> semijoin(german, risk1, on = :id)
+6×5 DataFrame
+ Row │ id     Age    Sex     Job    Housing
+     │ Int64  Int64  String  Int64  String
+─────┼──────────────────────────────────────
+   1 │     4     53  male        2  free
+   2 │     5     35  male        1  free
+   3 │     6     53  male        2  own
+   4 │     7     35  male        3  rent
+   5 │     8     61  male        1  own
+   6 │     9     28  male        3  own
+```
+
+The below example will return true, demonstrating that a `semijoin` is the same as `innerjoin` with 
+only columns from `german` data frame.
+
+```jldoctest dataframe
+julia> semijoin(german, risk, on = :id) == innerjoin(german, risk, on = :id)[:, names(german)]
+true
+```
+
+## Cross Join
+
+The output is the cartesian product of rows from the first (left) and second (right) arguments to join.
+
+Now, let's have a look on `crossjoin`:
+
+```jldoctest dataframe
+julia> crossjoin(german, risk, makeunique=true);
+```
+
+In the above example, A `crossjoin` took all the rows from `german` data frame and for each row, it matches it 
+to every single row of `risk` data frame. At first impression, this might not make any sense, but it’s a great 
+method for finding all combinations of 2 data frames. Our new data frame has 100 rows.
+
+!!! note
+
+    Cross joins are the only kind of join that does not use a `on` key:
+    `crossjoin(persons, earnings, makeunique = true)`
+
+For more clarification, now let’s say we want to design a new data frame by changing the ingredients. To 
```

Comment:
Clarification of what?

Maybe also just add a comment here where these examples are taken from?